walk-in-interview/src/main/java/com/google/job/servlets/InterestedJobsServlet.java

```diff
@@ -0,0 +1,65 @@
+package com.google.job.servlets;
+
+import com.google.job.data.*;
+import com.google.utils.ServletUtils;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/** Servlet that handles adding/removing and getting an applicant's interested jobs. */
+@WebServlet("/my-interested-list")
+public final class InterestedJobsServlet extends HttpServlet {
+    private static final long TIMEOUT_SECONDS = 5;
+
+    private static final String PAGE_SIZE_PARAM = "pageSize";
+    private static final String PAGE_INDEX_PARAM = "pageIndex";
+
+    private JobsDatabase jobsDatabase;
+
+    @Override
+    public void init() {
+        this.jobsDatabase = new JobsDatabase();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        try {
+
+            int pageSize = JobsListingsServlet.parsePageIndex(request);
+            int pageIndex = JobsListingsServlet.parsePageIndex(request);
+            
+            JobPage jobPage = fetchJobPageDetails(pageSize, pageIndex);
+
+            String json = ServletUtils.convertToJsonUsingGson(jobPage);
+            response.setContentType("application/json;");
+            response.getWriter().println(json);
+        } catch(IllegalArgumentException | ServletException | ExecutionException | TimeoutException e) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
can you log the exception? here it's swallowed. specially the servlet or execution exception, we really want to know when the error comes from the server-side.